kafka/src/main/java/io/micronaut/configuration/kafka/intercept/KafkaClientIntroductionAdvice.java

```diff
@@ -368,17 +368,116 @@ public void onComplete() {
+
+            //noinspection SubscriberImplementation
+            send.subscribe(new Subscriber() {
+                boolean completed = false;
+
+                @Override
+                public void onSubscribe(Subscription s) {
+                    s.request(1);
+                }
+
+                @Override
+                public void onNext(Object o) {
+                    sendResult.complete(o);
+                    completed = true;
+                }
+
+                @Override
+                public void onError(Throwable t) {
+                    sendResult.completeExceptionally(wrapException(context, t));
+                }
+
+                @Override
+                public void onComplete() {
+                    if (!completed) {
+                        // empty publisher
+                        sendResult.complete(null);
+                    }
+                }
+            });
+            return sendResult;
+        } else {
+
+            ProducerRecord record = buildProducerRecord(context, producerState, value);
+            if (LOG.isTraceEnabled()) {
+                LOG.trace("@KafkaClient method [" + logMethod(context) + "] Sending producer record: " + record);
             }
-        }));
+
+            boolean transactional = producerState.transactional;
+            Producer<?, ?> kafkaProducer = producerState.kafkaProducer;
+            try {
+                if (transactional) {
+                    LOG.trace("Beginning transaction for producer: {}", producerState.transactionalId);
+                    kafkaProducer.beginTransaction();
+                }
+
+                Supplier<Object> sendSupplier = () ->  kafkaProducer.send(record, (metadata, exception) -> {
+                    System.out.println("IN SEND COMPLETION CALLBACK");
+                    if (exception != null) {
+                        sendResult.completeExceptionally(wrapException(context, exception));
+                    } else {
+                        if (returnType.equalsType(Argument.VOID_OBJECT)) {
+                            sendResult.complete(null);
+                        } else {
+                            Optional<?> converted = conversionService.convert(metadata, returnType);
+                            if (converted.isPresent()) {
+                                sendResult.complete(converted.get());
+                            } else if (returnType.getType() == producerState.bodyArgument.getType()) {
+                                sendResult.complete(value);
+                            }
+                        }
+                    }
+                });
+                CompletableFuture<Object> delayedResultFuture = null;
+                if (producerState.executorService != null) {
+                    delayedResultFuture = CompletableFuture.supplyAsync(sendSupplier, producerState.executorService).thenCompose(o -> sendResult);
+                    System.out.println("SEND HAS BEEN INVOKED ASYNCHRONOUSLY");
```

Comment:
Actually, what's even worse is that this is in an alternate version of the returnCompletableFuture method (from before I move all of the async logic up into `intercept`) that shouldn't even be included in the PR. I was keeping it around in case I needed to go back to it, and then forgot to clean it out. 🤦‍♂️ Thanks for catching it.

Reply:
Fixed in [c9b2b93](https://github.com/micronaut-projects/micronaut-kafka/pull/955/commits/c9b2b93a3bf73302f98b66430a44239ad4e87e3d)